src/migration/service/migrator.ts

```diff
@@ -57,3 +85,29 @@ export async function resolveSqlMigrations(migrationPath: string): Promise<SqlMi
 
   return Promise.all(migrationPromises);
 }
+
+/**
+ * Resolve all the migration source for each of the migration entries using the configurations.
+ *
+ * @param {string} migrationPath
+ * @param {string} extension
+ * @returns {Promise<SqlMigrationEntry[]>}
+ */
+export async function resolveJavaScriptMigrations(
+  migrationPath: string,
+  extension: string = FileExtensionTypes.JS
+): Promise<JavaScriptMigrationEntry[]> {
+  const migrationNames = await getJavaScriptMigrationNames(migrationPath, extension);
+  const migrationPromises = migrationNames.map(async name => {
+    const filename = `${name}.${extension}`;
+
+    const { up, down } = require(path.resolve(migrationPath, filename));
```

Comment:
You should log here both before and after to allow better debugging.

Reply:
Also, does `require` work automatically for TS files? I don't think so, might need a ts importer of some sort eg: ts-node's lib.

Reply:
Not sure. But he has included the screenshots. @samirsilwal Did you try running this for bot `.ts` migration files and `.js` files?

Reply:
I remember us doing `import module = require("module")`. Isn't this similar? It might work actually. 💥

Reply:
I still don't think it works for ts files. The script runs as a JS binary, how can it import a TS file in the run-time and interpret TS w/o a transpiler. The plain require shouldn't work here, unless the ts-node is invoked someway.

Reply:
I tested it both for js and ts migrations its working for both in same way.

Reply:
Can you also put the TS migration files code? You've used static types and typescript specifics there?

Reply:
![Screenshot from 2021-03-03 10-24-16](https://user-images.githubusercontent.com/35683880/109753598-a7b36080-7c0a-11eb-837d-023cb791f88b.png)

here is users table migration

Reply:
How did you run sync-db, is it using the binary? Pretty strange that it works with `require`.

Reply:
found the issue after debugging with @mesaugat the way i was testing creating workspaces ran the migrations in ts as well.Testing after creating a zip failed to run ts migration. cc @kabirbaidhya

Reply:
Solved the issue with ts-node register use in `migrator.ts`

Reply:
Understood. You should always test it with the js binary after you build the source - this will clear such confusions.